src/psbt.js

```diff
@@ -231,6 +234,135 @@ export function psbtOutputFormatter(output) {
   }
 }
 
+/**
+ * Translates a PSBT into inputs/outputs consumable by non-PSBT devices in the
+ * `unchained-wallets` library.
+ *
+ * FIXME - Have only confirmed this is working for P2SH addresses on Ledger on regtest
+ *
+ * @param {module:networks.NETWORKS} network - bitcoin network
+ * @param {module:multisig.MULTISIG_ADDRESS_TYPES} addressType - address type
+ * @param {String} psbt - PSBT as a base64 or hex string
+ * @param {Object} keyDetails - Object containing signing key details (Fingerprint + bip32 prefix)
+ * @returns {null|Object} returns object with the format
+ * {
+ *    txInputs: [],
+ *    txOutputs: [],
+ *    bip32Derivations: [],
+ * }
+ */
+export function translatePSBT(network, addressType, psbt, keyDetails) {
+  let localPSBT = autoLoadPSBT(psbt, {network: networkData(network)});
+  if (localPSBT === null) return null;
+
+  let bip32Derivations = [];
+  // To provide the non-PSBT wallet with the proper info, we actually
+  // need to grab some info from the "data inputs" and some other
+  // information from the "transaction inputs".
+  let txInputs = localPSBT.txInputs.map((input, index) => {
```

Comment:
Any way to split this up like the `addSignatureToPSBT` for readability?

Reply:
open to suggestions - I tried to do a couple things but ended up reverting back to this. Getting the necessary pieces from the right place is a little bit tangled. I think the comments help but ofc I am biased.

Reply:
A rough draft of something that might work: https://github.com/humanumbrella/unchained-bitcoin/pull/2. Kind of related to the previous comment about the `txDetails` (?) below. breaking these out into more functional pieces might help. There are some loops that could be consolidated, but that's not unusual in functional stuff from what I've seen (e.g. chained things like `arr.filter(...).map(...)`.

Reply:
closed that one and edited it a bit - lmk